fighthealthinsurance/views.py

```
@@ -1447,8 +1514,19 @@ def post(self, request):
             for k, v in elems.items():
                 key = k
+                # Per-key guard: one unmappable key (questions regenerated
+                # between render and submit -> IndexError, malformed key ->
+                # ValueError) must lose only ITS mapping, not silently drop
+                # every answer the user just typed -- the old whole-loop try
+                # discarded the entire questionnaire on the first bad key.
                 if "appeal_generated_" in k:
-                    key = generated_questions[int(k.split("_")[-1]) - 1][0]
+                    try:
+                        key = generated_questions[int(k.split("_")[-1]) - 1][0]
+                    except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)